src/GovUk.Education.ExploreEducationStatistics.Admin/ViewModels/Public.Data/BatchMappingUpdateViewModels.cs

```diff
@@ -0,0 +1,80 @@
+#nullable enable
+using System.Collections.Generic;
+using System.Linq;
+using FluentValidation;
+using GovUk.Education.ExploreEducationStatistics.Common.Model.Data;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Model;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.ViewModels.Public.Data;
+
+public record BatchLocationMappingUpdatesRequest
+{
+    public List<LocationMappingUpdateRequest> Updates { get; init; } = [];
+
+    public class Validator : AbstractValidator<BatchLocationMappingUpdatesRequest>
+    {
+        public Validator()
+        {
+            RuleForEach(request => request.Updates)
+                .SetValidator(new LocationMappingUpdateRequest.Validator());
+        }
+    }
+}
+
+public record LocationMappingUpdateRequest
+{
+    public GeographicLevel? Level { get; init; }
+
+    public string? SourceKey { get; init; } = null;
+
+    public string? CandidateKey { get; init; }
+
+    public MappingType? Type { get; init; }
```

Comment:
Would `None` never be valid here? Wonder if we should add that to the validator below. I think it should only be in `AutoNone`, `AutoMapped`, `ManualNone` or `ManualMapped` when it is coming through to this endpoint right?

Reply:
> when this is `AutoMapped` or `AutoNone`, is it just the user confirming that the automapping was correct?

I flipped and flopped over this one.  Initially it would only be `Manual*` that we'd expect as the user worked through the form, but I envisaged the possibility of a user "undoing" a manual decision they'd made, and returning it back to "Auto*".

Tbh since the UI designs don't currently allow for this, I'll just accept manual mappings for now, and it's not hard to change things down the line if we need to.

Reply:
> Would `None` never be valid here? Wonder if we should add that to the validator below. I think it should only be in `AutoNone`, `AutoMapped`, `ManualNone` or `ManualMapped` when it is coming through to this endpoint right?

As per my previous reply, I'm changing this to only accept "Manual*" for now.

Reply:
Done, and added some more useful error messages as well to help out the front end a bit more.

Reply:
Ah, to be honest you make a good point about the user changing their minds and changing it back! I'm a bit torn now.

But I must agree, at first I was expecting only `Manual*` to be accepted. So I'll approve as-is and we can always change it later if needs be :)